core-bundle/contao/modules/ModulePersonalData.php

```diff
@@ -307,6 +308,11 @@ protected function compile()
 						// Set the new field in the member model
 						$blnModified = true;
 						$objMember->$field = $varValue;
+
+						if ($objWidget instanceof FormPassword)
```

Comment:
not sure this is the correct check. The session should be migrated if any of the security credentials change. This would include the username. I would think since there is no harm in migrating the session, we could simply just do it whenever the personal data is updated?

Reply:
> we could simply just do it whenever the personal data is updated?

I don‘t think that is necessary. We should only migrate the session ID if the username or the password changes. I will adjust the PR accordingly.

Reply:
Changed in 4fdf2db5d6144e73b0bdb468e941cad26b00ac68.